variables.tf

```diff
@@ -183,3 +182,34 @@ variable "disable_mfa" {
   type        = bool
   default     = false
 }
+
+variable "disabled" {
+  description = "Specifies whether the user is disabled, which prevents logging in and aborts all the currently-running queries for the user."
+  type        = bool
+  default     = false
+}
+
+variable "naming_scheme" {
+  description = <<EOT
+  Naming scheme configuration for the resource. This configuration is used to generate names using context provider:
+    - `properties` - list of properties to use when creating the name - is superseded by `var.context_templates`
+    - `delimiter` - delimited used to create the name from `properties` - is superseded by `var.context_templates`
+    - `context_template_name` - name of the context template used to create the name
+    - `replace_chars_regex` - regex to use for replacing characters in labels created by the provider - any characters that match the regex will be removed from the name
+    - `extra_labels` - map of extra label-value pairs, used to create a name
+  EOT
+  type = object({
+    properties            = optional(list(string), ["environment", "name"])
+    delimiter             = optional(string, "_")
+    context_template_name = optional(string, "snowflake-user")
```

Comment:
what if we start using actual resources going forward? I think this could be less confusing for the end-users.
`snowflake_user`, `aws_s3_bucket`, etc?

Reply:
It makes sens, sort of - I decided to comply with previous defaults `descriptor_name`.
In the same time, that will sometimes not be a resource_name - example `snowflake-project-role` - so maybe this kind of differentiations makes sense